component/loki/source/heroku/heroku.go

```diff
@@ -30,38 +30,13 @@ func init() {
 // Arguments holds values which are used to configure the loki.source.heroku
 // component.
 type Arguments struct {
-	HerokuListener       ListenerConfig      `river:"listener,block"`
+	Server               *lhttp.ServerConfig `river:",squash"`
 	Labels               map[string]string   `river:"labels,attr,optional"`
 	UseIncomingTimestamp bool                `river:"use_incoming_timestamp,attr,optional"`
 	ForwardTo            []loki.LogsReceiver `river:"forward_to,attr"`
 	RelabelRules         flow_relabel.Rules  `river:"relabel_rules,attr,optional"`
 }
 
-// ListenerConfig defines a heroku listener.
-type ListenerConfig struct {
-	ListenAddress string `river:"address,attr,optional"`
-	ListenPort    int    `river:"port,attr"`
-	// TODO - add the rest of the server config from Promtail
-}
-
-// DefaultListenerConfig provides the default arguments for a heroku listener.
-var DefaultListenerConfig = ListenerConfig{
-	ListenAddress: "0.0.0.0",
-}
-
-// UnmarshalRiver implements river.Unmarshaler.
-func (lc *ListenerConfig) UnmarshalRiver(f func(interface{}) error) error {
-	*lc = DefaultListenerConfig
```

Comment:
Removing this default should have no effect, because we would be passing an empty string as an address to weaveworks server, which has the same meaning as `0.0.0.0` - is my understanding correct here?

Reply:
Yup, weavewowrks applies no default, hence the resulting address in the [`net.Listen`](https://pkg.go.dev/net#Listen) call to get the listen socket is `:PORT`. AFAIK an empty listen address is the same as `0.0.0.0`, which is listening for every IP address.